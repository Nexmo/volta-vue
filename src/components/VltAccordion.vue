<template>
  <div class="Vlt-accordion" :class="classObject">
    <div :id="id" class="Vlt-accordion__content" v-if="styled">
      <button class="Vlt-accordion__trigger" @click="toggleAccordion()">
          {{label}}
      </button>
      <transition name="Vlt-accordion__panel"
        @enter="enter"
        @after-enter="afterEnter"
        @leave="leave"
      >
      <div class="Vlt-accordion__content_wrap" v-if="styled && showAccordion">
          <slot></slot>
      </div>
      </transition>
    </div>
    <slot v-if="trigger || panel"></slot>
  </div>
</template>

<style lang="scss">
    .Vlt-accordion__content {
      height: auto !important;
    }
    .Vlt-accordion__panel-enter, .Vlt-accordion__panel-leave-to {
      height: 0;
    }
    .Vlt-accordion__panel-enter-active, .Vlt-accordion__panel-leave-active {
      overflow: hidden;
      transition: all .5s ease;
    }
</style>

<script>

export default {
  name: 'vlt-accordion',

  props: {
    app: {
      type: Boolean,
      default: false,
    },
    id: {
      type: String,
      required: false,
    },
    label: {
      type: String,
      required: false,
    },
    panel: {
      type: Boolean,
      default: false
    },
    trigger: {
      type: Boolean,
      default: false
    }
  },

  computed: {
    classObject() {
      const obj = {
        'Vlt-js-accordion__trigger': this.trigger,
        'Vlt-js-accordion__content': this.panel,
        ['Vlt-accordion-' + this.label]: !this.styled,
      };
      return obj;
    },
  },

  data() {
    return {
      showAccordion: false,
      styled: !this.trigger && !this.panel,
    }
  },

  methods: {
    enter(element) {
      const width = getComputedStyle(element).width;

      element.style.width = width;
      element.style.position = 'absolute';
      element.style.visibility = 'hidden';
      element.style.height = 'auto';

      const height = getComputedStyle(element).height;

      element.style.width = null;
      element.style.position = null;
      element.style.visibility = null;
      element.style.height = 0;

      getComputedStyle(element).height;//force repaint

      setTimeout(() => {
        element.style.height = height;
      });
    },
    afterEnter(element) {
      element.style.height = 'auto';
    },
    leave(element) {
      const height = getComputedStyle(element).height;
      
      element.style.height = height;

      getComputedStyle(element).height;//force repaint

      setTimeout(() => {
        element.style.height = 0;
      });
    },
    toggleAccordion() {
      if (this.styled) {
        this.showAccordion = !this.showAccordion;
      }
    },
  },
};
</script>