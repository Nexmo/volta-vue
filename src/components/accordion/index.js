export { default as VltAccordion } from './VltAccordion';
export { default as VltAccordionItem } from './VltAccordionItem';
