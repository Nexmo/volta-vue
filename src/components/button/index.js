import VltButton from './VltButton';
import VltGroupButton from './VltGroupButton';

export { VltButton, VltGroupButton };
