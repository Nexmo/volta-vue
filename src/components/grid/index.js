import VltCol from './VltCol';
import VltGrid from './VltGrid';

export {
  VltCol,
  VltGrid,
};
