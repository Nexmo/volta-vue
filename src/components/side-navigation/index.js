import VltSidemenu from './VltSidemenu';
import VltSidemenuItem from './VltSidemenuItem';
import VltSidemenuLink from './VltSidemenuLink';
import VltSidemenuTitle from './VltSidemenuTitle';
import VltSidemenuTrigger from './VltSidemenuTrigger';

export { VltSidemenu, VltSidemenuItem, VltSidemenuLink, VltSidemenuTitle, VltSidemenuTrigger };
