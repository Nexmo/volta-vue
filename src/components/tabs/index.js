import VltTab from './VltTab';
import VltTabs from './VltTabs';

export { VltTab, VltTabs };
