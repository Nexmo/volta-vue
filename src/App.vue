<template>
  <div id="app">
  	<h2>Volta Vue.js API</h2>
  	<div v-for="component in components" class="Vlt-card">
  		<h3 class="Vlt-title--icon" :id="component.title"><vlt-icon :icon="component.icon" />{{component.title}} - {{component.html}}</h3>

      <vlt-table v-if="component.propertyRows" :columns="propertyColumns" :rows="component.propertyRows">
      	<template slot="item" slot-scope="slotProps">
      		<td>{{ slotProps.item.property }}</td>
      		<td>{{ slotProps.item.type }}</td>
      		<td><code>{{ slotProps.item.default }}</code></td>
      	</template>
      </vlt-table>

      <vlt-table v-if="component.eventRows" :columns="eventColumns" :rows="component.eventRows">
      	<template slot="item" slot-scope="slotProps">
      		<td>{{ slotProps.item.event }}</td>
      		<td>{{ slotProps.item.description }}</td>
      	</template>
      </vlt-table>
  	</div>
	</div>
</template>

<script>
  import { VltIcon, VltTable } from './components';
  import VoltaIcons from '@vonagevolta/core/dist/symbol/volta-icons.svg';

  export default {
    name: 'app',

    components: {
      VltIcon,
      VltTable
    },

    data () {
      return { 
      	components: [
      		{
      			title: 'Badge',
      			html: 'vlt-badge',
      			icon: 'bookmark',
      			propertyRows: [
		        	{ property: 'color', type: 'String' }, 
		        	{ property: 'dismissable', type: 'Boolean', default: true }
		        ],
      			eventRows: [
		        	{ event: 'dismissed', description: 'when the badge is dismissed' }
		        ]
      		},
      		{
      			title: 'Button',
      			html: 'vlt-button',
      			icon: 'circle-nested',
      			propertyRows: [
		        	{ property: 'app', type: 'Boolean', default: false }, 
		        	{ property: 'destructive', type: 'Boolean', default: true },
		        	{ property: 'disabled', type: 'Boolean', default: true },
		        	{ property: 'icon', type: 'String' },
		        	{ property: 'large', type: 'Boolean', default: true },
		        	{ property: 'primary', type: 'Boolean', default: true },
		        	{ property: 'secondary', type: 'Boolean', default: true },
		        	{ property: 'small', type: 'Boolean', default: true },
		        	{ property: 'spin', type: 'Boolean', default: true },
		        	{ property: 'tooltip-title', type: 'String' }
		        ]
      		},
      		{
      			title: 'Checkbox',
      			html: 'vlt-checkbox',
      			icon: 'checkbox',
      			propertyRows: [
		        	{ property: 'label', type: 'String' }, 
		        	{ property: 'options', type: 'Array' },
		        	{ property: 'property', type: 'String' },     	
		        	{ property: 'tooltip-position', type: 'String', default: 'left' },		        	
		        	{ property: 'tooltip-title', type: 'String' }
		        ]
      		},
      		{
      			title: 'Dropdown',
      			html: 'vlt-dropdown',
      			icon: 'circle-down',     			
		        propertyRows: [
		        	{ property: 'checked', type: 'Boolean' }, 
		        	{ property: 'id', type: 'String' },
		        	{ property: 'label', type: 'String' },     	
		        	{ property: 'selected', type: 'String', default: 'the first option in the options array' },		        	
		        	{ property: 'show-selection', type: 'Boolean' }
		        ]
      		},
      		{
      			title: 'Field',
      			html: 'vlt-field',
      			description: "Wraps an input element",
      			icon: 'image',     			
		        propertyRows: [
		        	{ property: 'big', type: 'Boolean', default: false }, 
		        	{ property: 'elastic', type: 'Boolean', default: false },
		        	{ property: 'error-message', type: 'String' },     	
		        	{ property: 'label', type: 'String' },		        	
		        	{ property: 'optional', type: 'Boolean', default: false }
		        ]
      		},
      		{
      			title: 'Icon',
      			html: 'vlt-icon',
      			icon: 'info',     			
		        propertyRows: [
		        	{ property: 'icon', type: 'String' }, 
		        	{ property: 'clickable', type: 'Boolean', default: false },
		        	{ property: 'pointer', type: 'Boolean', default: false },     	
		        	{ property: 'spin', type: 'Boolean', default: false },		        	
		        	{ property: 'small', type: 'Boolean', default: false }
		        ]
      		},
      		{
      			title: 'Input',
      			html: 'vlt-input',
      			icon: 'type',     			
		        propertyRows: [
		        	{ property: 'disabled', type: 'Boolean', default: false  }, 
		        	{ property: 'hint', type: 'String' },
		        	{ property: 'id', type: 'String' },     	
		        	{ property: 'label', type: 'String' },		        	
		        	{ property: 'max', type: 'Number' },       			        	
		        	{ property: 'maxlength', type: 'Number' }, 			        	
		        	{ property: 'min', type: 'Number' },	        				        	
		        	{ property: 'placeholder', type: 'String' },        	 				        	
		        	{ property: 'textarea', type: 'Boolean', default: false },		        				        	
		        	{ property: 'type', type: 'String', default: "text" },	        		        				        	
		        	{ property: 'val', type: 'String | Number' }
		        ]
      		},
      		{
      			title: 'Modal',
      			html: 'vlt-modal',
      			icon: 'modal',   
		        propertyRows: [
		        	{ property: 'can-canel', type: 'Boolean', default: true  }, 
		        	{ property: 'cancel-text', type: 'String' },
		        	{ property: 'confirm-btn-icon', type: 'String' }, 
		        	{ property: 'destructive', type: 'Boolean', default: false }, 
		        	{ property: 'dynamic', type: 'Boolean', default: false }, 
		        	{ property: 'extra-btn-class', type: 'String' }, 
		        	{ property: 'extra-btn-icon', type: 'String' }, 
		        	{ property: 'extra-btn-label', type: 'String' }, 
		        	{ property: 'id', type: 'String' }, 
		        	{ property: 'large', type: 'Boolean', default: false }, 
		        	{ property: 'message', type: 'String' }, 
		        	{ property: 'notify', type: 'Boolean', default: false }, 
		        	{ property: 'ok-text', type: 'String' }, 
		        	{ property: 'title', type: 'String' },  
		        	{ property: 'visible', type: 'Boolean', default: false }, 
		        ],
		        eventRows: [
		        	{ event: 'close', description: 'when the modal is cancelled' },
		        	{ event: 'confirm', description: 'when the modal is confirmed' },	        	
		        	{ event: 'extraBtnClick', description: 'when the extra button is clicked' }
		        ]
      		},
      		{
      			title: 'Radio',
      			html: 'vlt-radio',
      			icon: 'check-circle',   			
		        propertyRows: [
		        	{ property: 'checked', type: 'Boolean', default: false  }, 
		        	{ property: 'hint', type: 'String' }, 
		        	{ property: 'inline', type: 'Boolean', default: false  },
		        	{ property: 'label', type: 'String' }, 
		        	{ property: 'name', type: 'String' }, 
		        	{ property: 'val', type: 'String' }
		        ]
		    },
      		{
      			title: 'Switch',
      			html: 'vlt-switch',
      			icon: 'switch',   			
		        propertyRows: [
		        	{ property: 'checked', type: 'Boolean', default: false  },
		        	{ property: 'red', type: 'Boolean', default: false },
		        	{ property: 'small', type: 'Boolean', default: false  }
		        ]
      		},
      		{
      			title: 'Table',
      			html: 'vlt-table',
      			icon: 'table-1',  	
		        propertyRows: [ 
		        	{ property: 'columns', type: 'Array' },
		        	{ property: 'current-page', type: 'Number', default: 1 },
		        	{ property: 'pagination', type: 'Boolean', default: false  },
		        	{ property: 'page-size', type: 'Number', default: 20  },
		        	{ property: 'rows', type: 'Array' },
		        	{ property: 'total', type: 'Number', default: 0 }
		        ]
      		},
      		{
      			title: 'Tooltip',
      			html: 'vlt-tooltip',
      			icon: 'tooltip',
		        propertyRows: [
		        	{ property: 'position', type: 'String', default: "left"  },
		        	{ property: 'title', type: 'String' }
		        ]
      		}
      	],
      	eventColumns: [
      		{ title: 'Event' }, 
      		{ title: 'Description' }
      	],
      	propertyColumns: [
      		{ title: 'Property' },
      		{ title: 'Type' },
      		{ title: 'Default' }
  			],
      	VoltaIcons,
      	VltTable
      }
    }
  }
</script>

<style lang="scss">
$Vlt-font-url: '~@vonagevolta/core/fonts/';
@import '~@vonagevolta/core/scss/volta';

body {
  background: $grey-lighter;
  padding: $unit2 $unit3	
}
</style>
