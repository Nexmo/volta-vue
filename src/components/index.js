import VltBadge from './badge/VltBadge';
import { VltButton, VltGroupButton } from './button';
import VltCallout from './callout/VltCallout';
import VltCheckbox from './checkbox/VltCheckbox';
import { VltCol, VltGrid } from './grid';
import VltComposite from './composite/VltComposite';
import VltDropdown from './dropdown/VltDropdown';
import VltField from './field/VltField';
import VltFlash from './flash/VltFlash';
import VltIcon from './icon/VltIcon';
import VltInput from './input/VltInput';
import VltMargin from './margin/VltMargin';
import VltModal from './modal/VltModal';
import VltRadio from './radio/VltRadio';
import VltSelect from './select/VltSelect';
import { VltAccordion, VltAccordionItem } from './accordion';
import {
  VltSidemenu,
  VltSidemenuItem,
  VltSidemenuLink,
  VltSidemenuTitle,
  VltSidemenuTrigger,
} from './side-navigation';
import VltSpinner from './spinner/VltSpinner';
import VltSteps from './steps/VltSteps';
import VltSwitch from './switch/VltSwitch';
import { VltTab, VltTabs } from './tabs';
import VltTable from './table/VltTable';
import VltTitle from './title/VltTitle';
import VltTooltip from './tooltip/VltTooltip';
import VltGridList from './grid-list/VltGridList';

export {
  VltAccordion,
  VltAccordionItem,
  VltBadge,
  VltButton,
  VltCallout,
  VltCheckbox,
  VltCol,
  VltComposite,
  VltDropdown,
  VltField,
  VltFlash,
  VltGrid,
  VltGridList,
  VltGroupButton,
  VltIcon,
  VltInput,
  VltMargin,
  VltModal,
  VltRadio,
  VltSelect,
  VltSidemenu,
  VltSidemenuItem,
  VltSidemenuLink,
  VltSidemenuTitle,
  VltSidemenuTrigger,
  VltSpinner,
  VltSteps,
  VltSwitch,
  VltTab,
  VltTabs,
  VltTable,
  VltTitle,
  VltTooltip,
};
