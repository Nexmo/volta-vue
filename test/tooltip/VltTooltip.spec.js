import { shallowMount } from '@vue/test-utils';
import { VltTooltip } from '../../src/components';

describe('vlt-tooltip', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltTooltip, {
      propsData: {
        title: 'Tooltip Title',
      },
    });
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders correctly with text "Tooltip Test"', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
