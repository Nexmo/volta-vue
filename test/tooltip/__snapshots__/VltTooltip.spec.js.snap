// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-tooltip renders correctly with text "Tooltip Test" 1`] = `
<span
  class="Vlt-tooltip"
>
  <vlt-icon-stub
    class="Vlt-blue"
    clickable="true"
    icon="help-negative"
    small="true"
  />
</span>
`;
